tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/ExerciseBuilder.java

```diff
@@ -0,0 +1,122 @@
+package fi.helsinki.cs.tmc.langs;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.io.FileUtils;
+
+public class ExerciseBuilder {
+
+    private String beginSolution = "// BEGIN SOLUTION";
+    private String endSolution = "// END SOLUTION";
+    private String stubMarker = "// STUB:";
```

Comment:
The current ruby implementation allows any number of spaces where these have just one space. If we go with this implementation (which I think is reasonable), we'll have to audit our exercises for cases where there are extra spaces and change tmc-netbeans-author to match. Please make Trello tickets of these (and mark them sufficiently important that they won't be missed) unless you want to change this to match the current implementation.